kliff/trainer/kliff_trainer.py

```diff
@@ -0,0 +1,791 @@
+import hashlib
+import importlib
+import json
+import os
+import random
+from copy import deepcopy
+from datetime import datetime, timedelta
+from glob import glob
+from pathlib import Path
+from typing import Callable, Union
+
+import dill  # TODO: include dill in requirements.txt
+import numpy as np
+import yaml
+from loguru import logger
+
+import kliff.transforms.configuration_transforms
+from kliff._exceptions import TrainerError
+from kliff.dataset import Dataset
+from kliff.transforms.configuration_transforms import ConfigurationTransform
+from kliff.transforms.parameter_transforms import ParameterTransform
+from kliff.transforms.property_transforms import PropertyTransform
+
+from ..dataset.weight import Weight
```

Comment:
Probably OK to use relative import from the current directory, but let's change to use absolute import for all other cases like this one `..dataset.weight`